src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -147,22 +146,25 @@ object SparkDruidIndexer extends Logging
       )
 
     logInfo("Starting uniqes")
-    val partitionMap: Map[Long, Long] = baseData
-      .countApproxDistinctByKey(
-        0.05,
-        new DateBucketPartitioner(
-          dataSchema.getDelegate.getGranularitySpec.getSegmentGranularity,
-          passableIntervals
-        )
-      )
-      .map(
-        x => dataSchema.getDelegate.getGranularitySpec
-          .getSegmentGranularity
-          .truncate(new DateTime(x._1))
-          .getMillis -> x._2
+    var uniquesEvents = baseData
+    if (dataSchema.getDelegate.getParser.getParseSpec.getDimensionsSpec.hasCustomDimensions) {
+      val parseSpec = dataSchema.getDelegate.getParser.getParseSpec
+      val dims: Set[String] = (Set[String]() ++ parseSpec.getDimensionsSpec.getDimensions) -- parseSpec.getDimensionsSpec.getDimensionExclusions
```

Comment:
you can use `parseSpec.getDimensionsSpec.getDimensions.toSet` to make it more readable